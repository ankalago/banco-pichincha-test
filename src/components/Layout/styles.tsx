import styled from 'styled-components';

const TitleLayout = styled.h1`
  font-size: 1.5em;
  text-align: center;
  margin-bottom: 1em;
`;

const ContentLayout = styled.section`
  padding: 2em 1em;
`;

const GridLayout = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  column-gap: 1em;
  row-gap: 2em;

  .search {
    grid-column: span 3;
    @media (min-width: 768px) {
      grid-column: span 1;
    }
  }

  .add {
    grid-column-start: 4;
    grid-column-end: 6;
    @media (min-width: 768px) {
      grid-column-start: 5;
    }
  }
`

const TableLayout = styled.div`
  grid-column: span 5;
  margin-top: 1em;
  width: 100%;
  text-indent: 0;
  overflow-x: scroll;
  
  table {
    width: 100%;
  }

  th, td {
    display: table-cell;
    padding: 0 1rem 0.75rem 2rem;
    border-color: #e5e7eb;
    border-style: solid;
    border-bottom-width: 1px;
    font-size: 14px;
  }

  th {
    font-weight: 500;
  }

  td {
    padding-top: 1em;
    font-weight: 300;
  }
`

const FormLayout = styled.div`
  grid-column: span 5;
  border: 4px #e5e7eb dashed;
  padding: 1.5em;
  border-radius: 0.5rem;
`

export {
  TitleLayout,
  ContentLayout,
  GridLayout,
  TableLayout,
  FormLayout,
}
