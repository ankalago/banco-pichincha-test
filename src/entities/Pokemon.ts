export type Pokemon = {
  id?: number
  name: string
  image: string
  attack: number
  defense: number
  hp: number
  type: string
  idAuthor: number
}

export interface IFormPokemon {
  name: string;
  image: string;
  attack: number;
  defense: number;
  hp: number;
  type: string;
}
