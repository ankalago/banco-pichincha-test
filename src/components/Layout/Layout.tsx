import React from 'react';
import { ContentLayout, FormLayout, GridLayout, TableLayout, TitleLayout } from './styles';
import Input from '../Input/Input';
import Pokemons from '../Pokemons/Pokemons';
import Form from '../Form/Form';

type Props = {}

const Layout: React.FC<Props> = () => {

  return (
    <ContentLayout>
      <TitleLayout>List Pokemons</TitleLayout>
      <GridLayout>
        <Input name={"search"} label={"search"} placeholder="Search" />
        <button type="button" name="new" className="add">Nuevo</button>
        <TableLayout>
          <Pokemons />
        </TableLayout>
        <FormLayout>
          <Form/>
        </FormLayout>
      </GridLayout>
    </ContentLayout>
  )
}

export default Layout
