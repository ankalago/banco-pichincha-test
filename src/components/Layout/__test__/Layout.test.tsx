import { screen, render } from '@testing-library/react';
import Layout from '../Layout';
import { IFormValues } from '../../../entities/Pokemon';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();

const renderComponent = () =>
  render(
    <QueryClientProvider client={queryClient}>
      <Layout />
    </QueryClientProvider>,
  );

jest.mock('react-hook-form', () => ({
  ...jest.requireActual<IFormValues>('react-hook-form'),
  Controller: () => <></>,
  useForm: () => ({
    control: () => ({}),
    handleSubmit: () => jest.fn(),
    register: () => jest.fn(),
    formState: () => ({}),
    setValue: () => jest.fn()
  }),
}));

describe('LayoutComponent', () => {

  it('should render the component', () => {
    const { container } = renderComponent();
    const wrapper = screen.getByText(/List Pokemons/);
    expect(wrapper).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

})
