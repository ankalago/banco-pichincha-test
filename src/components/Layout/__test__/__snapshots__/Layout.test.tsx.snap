// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent should render the component 1`] = `
<div>
  <section
    class="sc-gswNZR equphs"
  >
    <h1
      class="sc-bcXHqe cUcCso"
    >
      List Pokemons
    </h1>
    <div
      class="sc-dkrFOg jxlyB"
    >
      <label
        for="search"
      >
        search
      </label>
      <div
        class="sc-jSUZER btAxHp"
      >
        <input
          data-testid="search"
          id="search"
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <button
        class="add"
        name="new"
        type="button"
      >
        Nuevo
      </button>
      <div
        class="sc-hLBbgP esEbUB"
      >
        <table>
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Image
              </th>
              <th>
                Attack
              </th>
              <th>
                Defense
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <div
        class="sc-eDvSVe hvxeMv"
      >
        <div
          class="sc-iBYQkv ksQtxZ"
        >
          New Pokemon
        </div>
        <form
          data-testid="form"
        >
          <div
            class="sc-ftTHYK dXNeFx"
          >
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="name"
              >
                Name
              </label>
              <div
                class="sc-jSUZER btAxHp"
              >
                <input
                  class=""
                  data-testid="name"
                  id="name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="hp"
              >
                Hard Power
              </label>
              <div
                class="sc-jSUZER btAxHp"
              >
                <input
                  class=""
                  data-testid="hp"
                  id="hp"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="type"
              >
                Type
              </label>
              <div
                class="sc-jSUZER btAxHp"
              >
                <input
                  class=""
                  data-testid="type"
                  id="type"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="image"
              >
                Image
              </label>
              <div
                class="sc-jSUZER btAxHp"
              >
                <input
                  class=""
                  data-testid="image"
                  id="image"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="attack"
              >
                Attack
              </label>
              <div
                class="sc-gKPRtg omiaA"
              >
                <span>
                  0
                </span>
                <input
                  class=""
                  data-testid="attack"
                  id="attack"
                  max="100"
                  min="0"
                  step="1"
                  type="range"
                  value="0"
                />
                <span>
                  100
                </span>
              </div>
            </div>
            <div
              class="sc-pyfCe eBIqyv"
            >
              <label
                for="defense"
              >
                Defense
              </label>
              <div
                class="sc-gKPRtg omiaA"
              >
                <span>
                  0
                </span>
                <input
                  class=""
                  data-testid="defense"
                  id="defense"
                  max="100"
                  min="0"
                  step="1"
                  type="range"
                  value="0"
                />
                <span>
                  100
                </span>
              </div>
            </div>
            <div
              class="sc-jrcTuL hlbgqN"
            >
              <button
                data-testid="submit"
                type="submit"
              >
                Guardar
              </button>
              <button
                data-testid="cancel"
                type="reset"
              >
                Cancelar
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </section>
</div>
`;
